diff --git a/node_modules/lodash.set/index.js b/node_modules/lodash.set/index.js
index 9f3ed6b..94894b3 100644
--- a/node_modules/lodash.set/index.js
+++ b/node_modules/lodash.set/index.js
@@ -505,10 +505,19 @@ function baseSet(object, path, value, customizer) {
       lastIndex = length - 1,
       nested = object;
 
+  function isUnsafeKey(key) {
+    return key === '__proto__' || key === 'constructor' || key === 'prototype';
+  }
+
   while (nested != null && ++index < length) {
     var key = toKey(path[index]),
         newValue = value;
 
+    // Patch: Prevent prototype pollution
+    if (isUnsafeKey(key)) {
+      return object;
+    }
+
     if (index != lastIndex) {
       var objValue = nested[key];
       newValue = customizer ? customizer(objValue, key, nested) : undefined;
